libsupport/include/katana/Cache.h

```diff
@@ -116,30 +117,26 @@ class KATANA_EXPORT Cache {
   }
 
   // Return a vector of node properties present in the cache, in LRU order
-  std::vector<std::string> GetNodeProperties() const {
+  std::vector<std::string> GetNodeProperties(
+      const std::string& rdg_prefix) const {
     std::vector<std::string> node_props;
     for (const auto& elt : lru_list_) {
-      if (elt.is_node()) {
+      if (elt.is_node() && elt.rdg_prefix() == rdg_prefix) {
```

Comment:
The cache key and value type is generic, yet we assume certain methods exist like `elt.is_node` and `elt.rdg_prefix` ? 

Options:
- Document this restriction, add tests to at least check the concept requirements
- Move this class closer to its user as it is clearly not generic enough to support arbitrary key or value types
- Use a more generic interface (key::operator==, Get, Put).

Reply:
Yeah, I hear you.  I started out with Cache.h as generic and PropertyCache as the place for node/edge specialization.  But a lot of PropertyCache has leaked into Cache.h and it isn't clear that Cache gains any value from being generic.

In addition, the cache's support for multiple RDGs is shakey and I need to be able to evict a specific entry, so I'll revisit the interface.